spec/system/distribution_system_spec.rb

```diff
@@ -0,0 +1,235 @@
+RSpec.describe "Distributions", type: :system do
+  before do
+    sign_in(@user)
+    @url_prefix = "/#{@organization.to_param}"
+
+    @partner = create(:partner, organization: @organization)
+    # allow_any_instance_of(StorageLocation).to receive(:geocode).and_return(true)
+    @storage_location = create(:storage_location, organization: @organization)
+    setup_storage_location(@storage_location)
+  end
+
+  it "User creates a new distribution" do
+    with_features email_active: true do
+      visit @url_prefix + "/distributions/new"
+
+      select @partner.name, from: "Partner"
+      select @storage_location.name, from: "From storage location"
+
+      fill_in "Comment", with: "Take my wipes... please"
+
+      expect do
+        click_button "Save", match: :first
+      end.to change { PartnerMailerJob.jobs.size }.by(1)
+
+      expect(page).to have_content "Distributions"
+      expect(page.find(".alert-info")).to have_content "reated"
+    end
+  end
+
+  it "User doesn't fill storage_location" do
+    visit @url_prefix + "/distributions/new"
+
+    select @partner.name, from: "Partner"
+    select "", from: "From storage location"
+
+    click_button "Save", match: :first
+    expect(page).to have_content "An error occurred, try again?"
+  end
+
+  context "With an existing distribution" do
+    let!(:distribution) { create(:distribution, :with_items, agency_rep: "A Person", organization: @user.organization) }
+
+    before do
+      visit @url_prefix + "/distributions"
+    end
+
+    it "the user can make changes to it" do
+      click_on "Edit", match: :first
+      expect do
+        fill_in "Agency representative", with: "SOMETHING DIFFERENT"
+        click_on "Save", match: :first
+        distribution.reload
+      end.to change { distribution.agency_rep }.to("SOMETHING DIFFERENT")
+    end
+
+    it "the user can reclaim it" do
+      expect do
+        accept_confirm do
+          click_on "Reclaim"
+        end
+        page.find ".alert"
+      end.to change { Distribution.count }.by(-1)
+      expect(page).to have_content "reclaimed"
+    end
+
+    context "when one of the items has been 'deleted'" do
+      it "the user can still reclaim it and it reactivates the item" do
+        item = distribution.line_items.first.item
+        item.destroy
+        expect do
+          accept_confirm do
+            click_on "Reclaim"
+          end
+          page.find ".alert"
+        end.to change { Distribution.count }.by(-1).and change { Item.count }.by(1)
+        expect(page).to have_content "reclaimed"
+      end
+    end
+  end
+
+  context "When creating a distribution and items have value" do
+    before do
+      item1 = create(:item, value: 10.5)
+      item2 = create(:item)
+      item3 = create(:item, value: 1)
```

Comment:
This was changed upstream and will cause the test to fail:
```suggestion
      item3 = create(:item, value_in_cents: 100)
```